app/views/devise/shared/_auth_nav.html.slim

```diff
@@ -0,0 +1,10 @@
+.clearfix.mxn1.mb1
+  - login_page = controller_name == 'sessions'
+  .sm-col.sm-col-6.px1.mb1
+    = link_to t('upaya.links.sign_up'), new_user_registration_path, \
+      class: 'btn btn-primary border-box col-12 center ' \
+             "#{login_page ? 'bg-gray' : 'bg-navy'}"
+  .sm-col.sm-col-6.px1.mb1
+    = link_to t('upaya.links.sign_in'), new_user_session_path, \
+      class: 'btn btn-primary border-box col-12 center ' \
+             "#{login_page ? 'bg-navy' : 'bg-gray'}"
```

Comment:
It's generally recommended to avoid logic inside views. For conditional classes, I usually create a helper method. For example:
```ruby
# app/helpers/link_tag_helper.rb
module LinkTagHelper
  def sign_up_link_tag(controller)
    link_to(
      t('upaya.links.sign_up'), 
      new_user_registration_path,
      class: 'btn btn-primary border-box col-12 center sign_up_bg_color(controller)'
    )
  end

  def sign_in_link_tag(controller)
    link_to(
      t('upaya.links.sign_in'), 
      new_user_session_path,
      class: 'btn btn-primary border-box col-12 center sign_in_bg_color(controller)'
    ) 
  end

  private

  def sign_up_bg_color(controller)
    return 'bg-gray' if controller == 'sessions'
    'bg-navy'
  end

  def sign_in_bg_color(controller)
    return 'bg-navy' if controller == 'sessions'
    'bg-gray'
  end
end
```
Then in the view, you would do:
```slim
.sm-col.sm-col-6.px1.mb1
  = sign_up_link_tag(controller_name)
.sm-col.sm-col-6.px1.mb1
  = sign_in_link_tag(controller_name)
```